chutney/action-impl/src/test/java/com/chutneytesting/action/kafka/KafkaBasicPublishActionTest.java

```diff
@@ -188,98 +160,90 @@ public void basic_publish_action_should_failed_when_timeout() throws Exception {
-
-      ActionExecutionResult actionExecutionResult = sut.execute();
-
-      assertThat(actionExecutionResult.status).isEqualTo(Success);
-
-      ConsumerRecord<Integer, String> singleRecord = KafkaTestUtils.getSingleRecord(consumer, TOPIC);
-      assertThat(singleRecord.value()).isEqualTo("my-test-value");
-
-      consumer.close();
-    }
-
-    @Test
-    public void producer_from_target_with_truststore_should_reject_ssl_connection_with_broker_without_truststore_configured() {
-      embeddedKafkaBroker.afterPropertiesSet();
-      Consumer<Integer, String> consumer = configureConsumer();
+        embeddedKafkaBroker.afterPropertiesSet();
+        try (Consumer<Integer, String> consumer = configureConsumer()) {
 
-      Target target = TestTarget.TestTargetBuilder.builder()
-          .withTargetId("kafka")
-          .withUrl("tcp://" + embeddedKafkaBroker.getBrokersAsString())
-          .withProperty("trustStore", KEYSTORE_JKS)
-          .withProperty("trustStorePassword", "server")
-          .withProperty("security.protocol", "SSL")
-          .build();
+            Target target = TestTarget.TestTargetBuilder.builder()
+                .withTargetId("kafka")
+                .withUrl("tcp://" + embeddedKafkaBroker.getBrokersAsString())
+                .build();
 
-      Map<String, String> props = new HashMap<>();
-      props.put("group.id", GROUP);
-      props.put("auto.commit.interval.ms", "10");
-      props.put("session.timeout.ms", "3000");
-      props.put("auto.offset.reset", "earliest");
+            Map<String, String> props = new HashMap<>();
+            props.put("group.id", GROUP);
 
-      Action sut = new KafkaBasicPublishAction(target, TOPIC, Map.of(), "my-test-value", props, null, logger);
+            Action sut = new KafkaBasicPublishAction(target, TOPIC, Map.of(), "my-test-value", props, null, logger);
 
-      ActionExecutionResult actionExecutionResult = sut.execute();
+            ActionExecutionResult actionExecutionResult = sut.execute();
 
-      assertThat(actionExecutionResult.status).isEqualTo(Failure);
+            assertThat(actionExecutionResult.status).isEqualTo(Success);
 
-      consumer.close();
+            ConsumerRecord<Integer, String> singleRecord = KafkaTestUtils.getSingleRecord(consumer, TOPIC);
+            assertThat(singleRecord.value()).isEqualTo("my-test-value");
+        }
     }
 
-    @Test
-    public void should_produce_message_to_broker_with_explicit_key() throws Exception {
+    @ParameterizedTest
+    @CsvSource(textBlock = """
+        /security/truststore.jks,truststore
+        /security/truststore_empty_pass.jks,'k'
+        /security/truststore_empty_pass.jks,null
+        """)
+    public void producer_from_target_with_truststore_should_reject_ssl_connection_with_broker_without_ssl_configured(String truststorePath, String truststorePass) throws URISyntaxException {
```

Comment:
same comment, strange on failing test case

Reply:
Same here. The embedded broker does not allow to do SSL.
On the other hand, i cleaned this test to check for the right failure in errors, i.e. `Send failed`.